tools/loader/ImageClassifier.cpp

```diff
@@ -99,9 +119,19 @@ void loadImagesAndPreprocess(const llvm::cl::list<std::string> &filenames,
   std::tie(imgHeight, imgWidth, isGray) = getPngInfo(filenames[0].c_str());
   const size_t numChannels = isGray ? 1 : 3;
 
-  // N x C x H x W
-  result->reset(ElemKind::FloatTy,
+  switch (imageLayout) {
+  case ImageLayout::NCHW:
+    result->reset(ElemKind::FloatTy,
                 {numImages, numChannels, imgHeight, imgWidth});
+    break;
+  case ImageLayout::NHWC:
+    result->reset(ElemKind::FloatTy,
+                {numImages, imgHeight, imgWidth, numChannels});
+    break;
+  default:
```

Comment:
remove `default` here, a compiler will ensure that case covers all values from `enum class`

Reply:
So I had not added the default as there was not a case where I could hit the default condition (the llvminit took care of this). However somebody else commented that I should add the llvm_unreachable as the default: So should I remove this or not?

Reply:
What was the motivation for llvm_unreachable?

The difference in behavior happens if you add a new enum option to the `enum class`.

On the one hand, If you do not have `default` option in the `switch` case you'll end up getting a compilation error that not all values are tested against in the `switch`. So, you'll have to fix this piece of code and add handling of the new enum value.

On the other hand, having `default` in place under the same circumstances will cause release build to ignore it and run with uninitialized result tensor (produce weird results). Debug build will fail with the assertion.

It feels that adding new layout should go along with the handling of it. Hence I'd prefer the first option.

Reply:
Thank you, I will remove it